src/player.cpp

```diff
@@ -13287,6 +13287,35 @@ void player::blossoms()
     }
 }
 
+int player::add_ammo_to_worn_quiver( item &ammo )
+{
+    std::vector<item *>quivers;
+    for( auto & worn_item : worn) {
+        if( worn_item.type->can_use( "QUIVER")) {
+            quivers.push_back( &worn_item);
+        }
+    }
+
+    // sort quivers by contents, such that empty quivers go last
+    std::sort( quivers.begin(), quivers.end());
```

Comment:
The comment is a lie. (That's why I don't like this kind of comment, you changed the code and suddenly the comment is out of sync.)

This sorts according to the default comparison of the items in `quivers`, in other words it sorts by comparing `item*` objects, which is simply pointer comparison and ignores the item data completely. And actually all item pointers come from one array (`player::worn`) and are added in order, so sorting it like this will do exactly nothing as it's already sorted.

The old code had the `sort_by_charges` struct to compare the item pointers by their content and used it accordingly when calling `std::sort`.

Minor nitpicking below: why revers the vector? Why not simply iterate through it backwards:
```C++
for( auto it = quivers.rbegin(); it != quivers.rend(); ++it ) {
    ...
}
```